lms/djangoapps/courseware/access.py

```diff
@@ -550,14 +554,18 @@ def can_load():
         students to see modules.  If not, views should check the course, so we
         don't have to hit the enrollments table on every module load.
         """
+        # Masquerading as a learner in a group needs to work in preview mode, and
+        # _has_staff_access_to_descriptor does not check group access if in preview mode.
+        if not _has_group_access(descriptor, user, course_key):
+            return ACCESS_DENIED
+
         if _has_staff_access_to_descriptor(user, descriptor, course_key):
             return ACCESS_GRANTED
 
         # if the user has staff access, they can load the module so this code doesn't need to run
```

Comment:
I find this comment confusing (not yours, I know). I think it refers to the line before, i.e. that the code is short circuited for staff. Maybe the comment should be moved there for clarity. These checks are then just for regular learners.